app/server.go

```diff
@@ -1625,100 +1625,157 @@ func (s *Server) SetLog(l *mlog.Logger) {
 
 func (a *App) GenerateSupportPacket() []model.FileData {
 	// If any errors we come across within this function, we will log it in a warning.txt file so that we know why certain files did not get produced if any
-	var warnings string
+	var warnings []string
 
 	// Creating an array of files that we are going to be adding to our zip file
 	fileDatas := []model.FileData{}
 
-	a.generateSupportPacketYaml(&warnings, &fileDatas)
-	a.createPluginsFile(&warnings, &fileDatas)
-	a.createSanitizedConfigFile(&warnings, &fileDatas)
-	a.getMattermostLog(&warnings, &fileDatas)
-	a.getNotificationsLog(&warnings, &fileDatas)
+	fileData, warning := a.generateSupportPacketYaml()
+
+	if warning == "" {
+		fileDatas = append(fileDatas, fileData)
+	} else {
+		warnings = append(warnings, warning)
+	}
```

Comment:
Sorry! This is a OOPSY on my end 😅 The idea was that if warning has text, we will ignore the return of fileData. The reason I had done it like this was so we don't have multiple return statements. Anyhow, I can see the confusion here and have refactored it to better fit my thinking. Sorry about the confusion.

Reply:
I've refactored it such that now function return a `(*fileData, string)` and then we check if `fileData != nil` instead of `warning == ""`. If `fileData != nil`, then something went wrong and therefore we append our warning to string array. Hope this makes more sense.